pkg/publicapi/test/tools.go

```diff
@@ -0,0 +1,54 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/bacalhau-project/bacalhau/pkg/publicapi/client"
+)
+
+const TimeToWaitForServerReply = 10 * time.Second
+const TimeToWaitForHealthy = 50 * time.Millisecond
+
+func WaitFor(ctx context.Context, c *client.APIClient, condition func(context.Context, *client.APIClient) bool) error {
```

Comment:
This is a helper wait method that is only used within the API package for testing purposes, and not a general purpose wait method. Returning a channel will just add more logic to each test case. Passing the durations as params will make each test define its duration and more durations will be sprinkled throughout the code, which is what you wanted to avoid. These default values are sufficient for the API server initialization, and we can explore defining different values if we add more conditions in the future